### ISSUE TITLE
RigRaymapHead leaks ground-truth cam2rig into its own prediction

### ISSUE THREAD

**Issue by engichang1467** [engichang1467/Open-Rig3R#38]
## Problem

`RigRaymapHead` is handed the ground-truth `cam2rig` and applies its rotation and
translation to its own prediction. The rig raymap *target* is built from that same
`cam2rig`. Part of the target is therefore given to the head for free during training.

`models/heads/rig_raymap_head.py:44-52`:

```python
R = cam2rig[..., :3, :3]                       # (B, V, 3, 3)
t = cam2rig[..., :3, 3] ...                    # (B, V, 3)
origins    = torch.einsum('bvij,bvnj->bvni', R, origins)
directions = torch.einsum('bvij,bvnj->bvni', R, directions)
if t is not None:
    origins = origins + t.unsqueeze(2)
```

`utils/raymap.py:99-102` builds the target from the same matrix:

```python
rotation    = cam2rig[..., :3, :3]
translation = cam2rig[..., :3, 3]
rig_directions = torch.einsum("bvij,bvpj->bvpi", rotation, directions)
rig_origins    = translation.unsqueeze(2).expand_as(rig_directions)
rig_raymap = torch.cat([rig_origins, rig_directions], dim=-1)
```

So the target origin is exactly `t`, and the head's predicted origin is
`R @ pred_origin + t`. A head that learns to output `pred_origin = 0` reproduces the
origin half of the target exactly, without learning anything about rig structure.

`models/rig3r.py:61` passes `cam2rig` in on every forward, so this is active during
training.

## Why it matters

The paper never wires extrinsics into the output. Rig metadata enters only as an additive
embedding on the patch tokens (§3.3), and is dropped 50% of the time (§3.4) precisely so
the model must infer rig structure from images. Hard-coding the transform means the rig
raymap loss is partly self-satisfying, and the rig-raymap term stops being evidence that
the model learned anything.

Rig-discovery *inference* is unaffected: `scripts/infer_rig_discovery.py:102` strips
`cam2rig` before the forward pass, so that eval path is clean. Training is the problem.

## Suggested fix

Drop the `cam2rig` argument from `RigRaymapHead.forward` and predict the rig raymap
directly from patch tokens, as the paper does. Remove the `cam2rig=` plumbing in
`models/rig3r.py:61` and `models/decoder_transformer.py`.

Rig pose should reach the model only through the metadata embedding path — see the
metadata-conditioning issue for that.
